src/utilities/piral-debug-utils/src/emulator.ts

```diff
@@ -58,6 +58,14 @@ export function withEmulatorPilets(requestPilets: PiletRequester, options: Emula
         console.error(`Requesting the pilets failed. We'll continue loading without pilets (DEBUG only).`, err);
         return [];
       })
-      .then((pilets) => appendix.then((debugPilets) => [...pilets, ...debugPilets]));
+      .then((pilets) =>
+        appendix.then((debugPilets) => {
+          const piletNames = debugPilets.reduce((piletNames, debugPilet) => {
+            piletNames.push(debugPilet.name);
+            return piletNames;
+          }, []);
+          return pilets.filter((pilet) => piletNames.indexOf(pilet.name) === -1).concat(debugPilets);
```

Comment:
Yeah I get that you want to filter this out, but I'd rather not have this logic on the client. Instead, I see this in the Piral CLI. And it can be optionally disabled via some CLI flag (e.g., `--no-unique-pilets`).

Reply:
You mean, pass from piral-cli (--no-unique-pilets) to `emulator.ts` (via process.env.NOUNIQUEPILETS), and then use this logic according to `process.env.NOUNIQUEPILETS`?

Reply:
No, don't use it in runtime code (i.e., `piral`, `piral-core`, or any lib like `piral-debug-utils`) - but in the piral-cli directly.

After all, this comes from the served pilet feed.

The logic is simple:

- By default pilets are debugged standalone, i.e., the usual feed is not used
- You can activate the usual feed, then you'll get all pilets in there (if you want to exclude certain ones, use the browser extension or browser debug API)
- If you want to use the usual feed always, but with more constraints, then use the `--feed` option
- You want to disable the unique pilet constraint on `--feed` then use `--no-unique-pilets`

Personally, I would not make the logic in `piral-debug-utils` too large. It should be quite lightweight and to the point.

Reply:
> You can activate the usual feed, then you'll get all pilets in there (if you want to exclude certain ones, use the browser extension or browser debug API)

Thanks for the guidance, this is a good solution.